From randy.dunlap@oracle.com  Wed Apr 30 11:22:01 2008
From: Randy Dunlap <randy.dunlap@oracle.com>
Date: Wed, 30 Apr 2008 09:01:17 -0700
Subject: sysfs: sysfs_update_group stub for CONFIG_SYSFS=n
To: torvalds <torvalds@linux-foundation.org>
Cc: akpm <akpm@linux-foundation.org>, jejb <James.Bottomley@HansenPartnership.com>, gregkh <greg@kroah.com>
Message-ID: <20080430090117.cb4a511e.randy.dunlap@oracle.com>

From: Randy Dunlap <randy.dunlap@oracle.com>

scsi_transport_spi uses sysfs_update_group() when CONFIG_SYSFS=n,
so provide a stub for it.

next-20080423/drivers/scsi/scsi_transport_spi.c:1467: error: implicit declaration of function 'sysfs_update_group'
make[3]: *** [drivers/scsi/scsi_transport_spi.o] Error 1

Signed-off-by: Randy Dunlap <randy.dunlap@oracle.com>
Cc: James Bottomley <James.Bottomley@HansenPartnership.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/sysfs.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/include/linux/sysfs.h
+++ b/include/linux/sysfs.h
@@ -190,6 +190,12 @@ static inline int sysfs_create_group(str
 	return 0;
 }
 
+static inline int sysfs_update_group(struct kobject *kobj,
+				const struct attribute_group *grp)
+{
+	return 0;
+}
+
 static inline void sysfs_remove_group(struct kobject *kobj,
 				      const struct attribute_group *grp)
 {
